File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int[] solution(int[] arr, int k) {
        int[] dis = IntStream.of(arr).distinct().toArray();
        int[] ans = new int[k];
        Arrays.fill(ans, -1);
        int len = Math.min(dis.length, ans.length);
        
        for (int i = 0; i < len; i++) {
            ans[i] = dis[i];
        }
        
        return ans;
    }
}

File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.kt
class Solution {
    fun solution(arr: IntArray, k: Int): IntArray {
        var distinct = arr.distinct().toIntArray()

        if (distinct.size == k) return distinct // 길이가 k 일때
        if (distinct.size > k) return distinct.sliceArray(0 until k) // k보다 길이가 길때
        
        var ans = IntArray(k) { -1 };
        for (i in 0 until distinct.size)
            ans[i] = distinct[i]

        return ans;
    }
}